analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/PubliclyWritableDirectoriesBase.cs

```diff
@@ -29,27 +29,26 @@ public abstract class PubliclyWritableDirectoriesBase<TSyntaxKind, TInvocationEx
         protected const string DiagnosticId = "S5443";
         private const string MessageFormat = "Make sure publicly writable directories are used safely here.";
         private const RegexOptions WindowsAndUnixOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant;
-        private readonly string[] linuxDirs =
-        {
-            "/dev/mqueue",
-            "/run/lock",
-            "/var/run/lock",
-        };
-        private readonly string[] macDirs =
+
+        protected static string[] InsecureEnvironmentVariables { get; }
+
+        private static readonly Regex UserProfile;
+        private static readonly Regex LinuxDirectories;
+        private static readonly Regex MacDirectories;
+        private static readonly Regex WindowsDirectories;
+        private static readonly Regex EnvironmentVariables;
+
+        static PubliclyWritableDirectoriesBase()
         {
-            "/var/tmp",
-            "/usr/tmp",
-            "/dev/shm",
-            "/library/caches",
-            "/users/shared",
-            "/private/tmp",
-            "/private/var/tmp",
-        };
-        private readonly Regex userProfile = new(@"^%USERPROFILE%[\\\/]AppData[\\\/]Local[\\\/]Temp", WindowsAndUnixOptions);
-        private readonly Regex linuxDirectories;
-        private readonly Regex macDirectories;
-        private readonly Regex windowsDirectories = new(@"^([a-z]:[\\\/]?|[\\\/][\\\/][^\\\/]+[\\\/]|[\\\/])(windows[\\\/])?te?mp([\\\/]|$)", WindowsAndUnixOptions);
-        private readonly Regex environmentVariables;
+            var insecureEnvironmentVariables = new[] { "tmp", "temp", "tmpdir" };
+            InsecureEnvironmentVariables = insecureEnvironmentVariables;
+            UserProfile = new("""^%USERPROFILE%[\\\/]AppData[\\\/]Local[\\\/]Temp""", WindowsAndUnixOptions);
+            LinuxDirectories = new($@"^({LinuxDirs().JoinStr("|", Regex.Escape)})(\/|$)", RegexOptions.Compiled);
```

Comment:
I would question the use of `RegexOptions.Compile` in general in this rule. The cost of creating in memory assembly with emitting instructions at runtime to evaluate regex is extremely high, often not worth the benefit of the speed bump. 

These regexes are generally straight-forward with not so much backtracking.

Did you try to measure the impact of the `Compiled` thing?

Reply:
This is not what the PR is about. The change shows significant improvements. Whether "Compiled" is bringing gains needs to be decided for all Regexes and a wide variety of projects. Create an issue, if you want us to look into it.